keras/layers/wrappers.py

```diff
@@ -94,6 +94,11 @@ class TimeDistributed(Wrapper):
     The input should be at least 3D, and the dimension of index one
     will be considered to be the temporal dimension.
 
+    Mask will be passed to the wrappered layer, and the output mask depends
+    on how it is handled in the wrappered layer:
+    1. reshape returned mask if any
+    2. collapse input mask (if any) to at most 2D, when no mask returned
```

Comment:
These two lines are not sufficiently descriptive of how masking is handled. Please rewrite in a clearer and more detailed way.